pandas/tests/series/indexing/test_setitem.py

```diff
@@ -945,3 +946,17 @@ def test_setitem_int_as_positional_fallback_deprecation():
     with tm.assert_produces_warning(FutureWarning, match=msg):
         ser3[4] = 99
     tm.assert_series_equal(ser3, expected3)
+
+
+def test_setitem_with_bool_indexer():
+    # GH#42530
+
+    df = DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    result = df.pop("b")
+    result[[True, False, False]] = 9
+    expected = Series(data=[9, 5, 6], index=range(3), name="b")
+    tm.assert_series_equal(result, expected)
+
+    df["a"][[True, False, False]] = 10
```

Comment:
What are you trying to do with chained assignment here?

Are you trying to do
```
df.loc[[True, False, False], "a"] = 10
```
?

Reply:
yes. Originally df["a"] (which becomes a series) was not throwing the error.
Got the idea from a comment in the bug report

Reply:
Please use the loc call then, chained assignment is not ideal